gremlin-go/driver/connection.go

```diff
@@ -63,8 +63,13 @@ func (connection *connection) write(request *request) (ResultSet, error) {
 	connection.logHandler.log(Info, writeRequest)
 	requestID := request.requestID.String()
 	connection.logHandler.logf(Info, creatingRequest, requestID)
-	connection.results[requestID] = newChannelResultSet(requestID)
-	return connection.results[requestID], connection.protocol.write(request)
+	resultSet := newChannelResultSet(requestID, connection.results)
+	connection.results[requestID] = resultSet
+	return resultSet, connection.protocol.write(request)
```

Comment:
nit, not sure if this breaks things
```suggestion
	connection.results[requestID] = newChannelResultSet(requestID, connection.results)
	return connection.results[requestID], connection.protocol.write(request)
```

Reply:
That could break things.

However infinitely unlikely it is, technically the read loop could finish reading the result and pluck it from the list before we return that.

Technically it's a race condition ever if it's basically impossible.